Sources/NIOEmbedded/Embedded.swift

```diff
@@ -183,7 +189,9 @@ public final class EmbeddedEventLoop: EventLoop {
         // Just drop all the remaining scheduled tasks. Despite having run all the tasks that were
         // scheduled when we entered the method this may still contain tasks as running the tasks
         // may have enqueued more tasks.
-        while self.scheduledTasks.pop() != nil {}
+        while let task = self.scheduledTasks.pop() {
+            task.fail(EventLoopError.shutdown)
+        }
```

Comment:
I _suspect_ this is a behavioural change. Can you write a test that encodes this?

Reply:
@Lukasa  So there is a test that is executing this `testFinishWithRecursivelyScheduledTasks` and without this change the test is actually crashing because the future is leaking. Which I am still not 100% sure why it did work before. Should I write an additional test that checks if the error is actually the `EventLoopError.shutdown`?

Reply:
Yes please!

Reply:
I found a different test that was already testing the `EmbeddedEventLoop` specifically when draining the remaining tasks. I just added an assertion that makes sure the last task is failed with the proper error. Is that fine with you?